hypothesis-python/src/hypothesis/database.py

```diff
@@ -50,6 +50,23 @@ def _usable_dir(path: Path) -> bool:
     return path.is_dir() and os.access(path, os.R_OK | os.W_OK | os.X_OK)
 
 
+def _usable_explicit_dir(path: os.PathLike) -> bool:
```

Comment:
Instead of creating a new function, let's just add the error-handling to `_usable_dir`.  It's pretty important that we go as far up the tree as necessary to find an existing path, not just one level - the user might have set a hypothesis home-directory which doesn't exist, but _can_ be created later if we need it.